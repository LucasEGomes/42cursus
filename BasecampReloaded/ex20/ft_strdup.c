#include <stdlib.h>

static int	ft_strlen(char *s)
{
	int	index;

	index = 0;
	while (s[index] != '\0')
		index++;
	return (index);
}

char	*ft_strdup(char *src)
{
	int		index;
	char	*result;

	index = 0;
	result = malloc(sizeof(*result) * (ft_strlen(src) + 1));
	if (result == NULL)
		return (NULL);
	while (src[index] != '\0')
	{
		result[index] = src[index];
		index++;
	}
	result[index] = '\0';
	return (result);
}
