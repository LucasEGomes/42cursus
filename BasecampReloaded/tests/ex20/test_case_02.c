#include "ex20.h"

int	g_fake = 0;

void	*__wrap_malloc(size_t size)
{
	if (g_fake)
		return (NULL);
	return (__real_malloc(size));
}

int	main(void)
{
	run("So she was considering in her own mind (as well as she could, ");
	g_fake = 1;
	run("for the hot day made her feel very sleepy and stupid), whether ");
	run("the pleasure of making a daisy-chain would be worth the trouble ");
	run("of getting up and picking the daisies, when suddenly a White ");
	g_fake = 0;
	run("Rabbit with pink eyes ran close by her.");
}
