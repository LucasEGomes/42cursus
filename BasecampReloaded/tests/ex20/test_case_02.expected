OK
Malloc failed
Malloc failed
Malloc failed
OK
